src/engine/Server.cpp

```diff
@@ -182,13 +182,34 @@ ad_utility::url_parser::ParsedRequest Server::parseHttpRequest(
         auto operation = ad_utility::url_parser::getParameterCheckAtMostOnce(
             parsedRequest.parameters_, paramName);
         if (operation.has_value()) {
-          parsedRequest.operation_ = Operation{operation.value()};
+          parsedRequest.operation_ = Operation{operation.value(), {}};
           parsedRequest.parameters_.erase(paramName);
         }
       };
+  auto addToDatasetClausesIfOperationIs = [&parsedRequest]<typename Operation>(
+                                              const std::string& key,
+                                              bool isNamed) {
+    if (Operation* op = std::get_if<Operation>(&parsedRequest.operation_)) {
+      ad_utility::appendVector(op->datasetClauses_,
+                               ad_utility::url_parser::parseDatasetClausesFrom(
+                                   parsedRequest.parameters_, key, isNamed));
+    }
+  };
+  auto addDatasetClauses = [&addToDatasetClausesIfOperationIs] {
+    addToDatasetClausesIfOperationIs.template operator()<Query>(
+        "default-graph-uri", false);
+    addToDatasetClausesIfOperationIs.template operator()<Query>(
+        "named-graph-uri", true);
+    addToDatasetClausesIfOperationIs.template operator()<Update>(
+        "using-graph-uri", false);
+    addToDatasetClausesIfOperationIs.template operator()<Update>(
+        "using-named-graph-uri", true);
+  };
```

Comment:
You can make this much more readable using the facilities in `util/TypeIdentity.h` (see the doc there, with a single `using namespace ...`, you can make the call basically
`addToDatasetClausesIf(ti<Query>, "default-graph-uri", false)`
which is much nicer than the `template operator()`

Reply:
Please also do the same for the `setOperationIfSpecifiedInParams` lambda that is used below.